### PR TITLE
xtimer: add support for arbitrary timer tick conversion

### PR DESCRIPTION
**Pull request by kaspar030**
(just an idea)

compare #3990.

This PR basically swaps XTIMER_SHIFT with the possiblity to use any conversion macro.

### CODE REVIEW

**Comment by gebart** [comment]
@kaspar030 Do you want to keep this open while #3990 is still open or can we close this?

File: sys/include/div.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#ifndef DIV_H
+#define DIV_H
+
+#include <stdint.h>
+
+static inline uint64_t div_u64_by_1000000(uint64_t val)
+{
+    return (uint64_t)(val * 0x431bde83) >> (0x12 + 32);
```
Comment (cgundogan) [inline sys/include/div.h]: should this also be `ul` for 16-bit and 8-bit platforms?

File: sys/include/div.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#ifndef DIV_H
+#define DIV_H
+
+#include <stdint.h>
+
+static inline uint64_t div_u64_by_1000000(uint64_t val)
```
Comment (cgundogan) [inline sys/include/div.h]: hm I like this approach. If it's really more effective then I will adopt the in RPL.

File: sys/include/div.h (hunk 1 above)
Comment (kaspar030) [inline sys/include/div.h]: which one? :)

Comment (cgundogan) [inline sys/include/div.h]: I was a little bit unsure how 8-bit or 16-bit platforms would interpret `0x431bde83`

Comment (gebart) [inline sys/include/div.h]: I have been thinking about this and #3990, I think the best solution may be
to convert the xtimer to use timer ticks internally like this PR, but to
use an opaque struct type to make the compiler error out mismatched timer
ticks vs microsecond constants.
I can't prepare a PR right now but after testing and working with #3990 for
some time that PR is beginning to feel unnecessarily convoluted and too
much overhead..

Comment (kaspar030) [inline sys/include/div.h]: (the div stuff is long merged, btw...)

**Comment by OlegHahm** [comment]
Still WIP => postpone.

**Comment by kYc0o** [comment]
Very old PR addressing a common issue. @kaspar030 would you like to close it in favor of #5608 ?

**Comment by miri64** [comment]
Fixes https://github.com/RIOT-OS/RIOT/issues/5237, but needs to be postponed :(

**Comment by kYc0o** [comment]
Wasn't solved with #5608 ?

**Comment by PeterKietzmann** [comment]
Yes, seems so.

**Pull request closed without merging**